aptos-indexer-processors-sdk/testing-framework/src/new_test_context.rs

```diff
@@ -0,0 +1,196 @@
+use std::fs;
+use std::path::Path;
+use aptos_protos::indexer::v1::TransactionsResponse;
+use aptos_indexer_processor_sdk::traits::processor_trait::ProcessorTrait;
+use aptos_protos::transaction::v1::Transaction;
+use async_trait::async_trait;
+use testcontainers::{
+    core::{IntoContainerPort, WaitFor},
+    runners::AsyncRunner,
+    ContainerAsync, GenericImage, ImageExt,
+};
+use url::Url;
+use aptos_indexer_processor_sdk::aptos_indexer_transaction_stream::TransactionStreamConfig;
+use crate::mock_grpc::MockGrpcServer;
+
+pub struct SdkTestContext<D: TestDatabase> {
+    pub transaction_batches: Vec<Transaction>,
+    pub database: D, // Holds the database setup (user-defined)
+}
+
+impl<D: TestDatabase> SdkTestContext<D> {
+    pub async fn new(txn_bytes: &[&[u8]], mut database: D) -> anyhow::Result<Self> {
+        let transaction_batches = txn_bytes
+            .iter()
+            .map(|txn| serde_json::from_slice(txn).unwrap())
+            .collect::<Vec<Transaction>>();
+
+        // Set up the database using the user-defined method
+        database.setup().await?;
+
+        Ok(SdkTestContext {
+            transaction_batches,
+            database,
+        })
+    }
+
+    /// Run the processor and pass user-defined validation logic
+    pub async fn run<F>(
+        &self,
+        processor: &impl ProcessorTrait,
+        txn_verion: u64,
+        generate_files: bool, // flag to control file generation
+        output_path: Option<String>, // Optional custom output path
+        verification_f: F,
+    ) -> anyhow::Result<serde_json::Value>
+    where
+        F: FnOnce(&str) -> anyhow::Result<serde_json::Value> + Send + Sync + 'static,
+    {
+        let transactions = self.transaction_batches.clone();
+        let transactions_response = vec![TransactionsResponse {
+            transactions,
+            ..TransactionsResponse::default()
+        }];
+
+        self.setup_mock_grpc(transactions_response, 1).await;
+
+        // Run the processor
+        processor.run_processor().await?;
+
+        // Pass the DB URL for user-defined validation
+        let db_url = self.database.get_db_url();
+        let db_values = verification_f(&db_url)?;
+
+        // Conditionally generate output files if the `generate_files` flag is true
+        if generate_files {
```

Comment:
Will there also be a diff mode added later?

Reply:
default mode will be a diff mode.